opennebula/resource_opennebula_virtual_network.go

```diff
@@ -501,215 +493,132 @@ func generateAR(armap map[string]interface{}, id int) string {
-			return fmt.Sprintf(arstring, arid, armac, arsize)
-		}
-		var arstring = `AR = [
-            AR_ID = %s,
-		    TYPE = IP6,
-		    SIZE = %d ]`
-		return fmt.Sprintf(arstring, arid, arsize)
-	}
-	if artype == "IP6_STATIC" {
-		if armac != "" {
-			var arstring = `AR = [
-                AR_ID = %s,
-		        TYPE = IP6_STATIC,
-		        MAC = %s,
-		        IP6 = %s,
-		        PREFIX_LENGTH = %s,
-		        SIZE = %d ]`
-			return fmt.Sprintf(arstring, arid, armac, arip6, arprefixlength, arsize)
-		}
-		var arstring = `AR = [
-            AR_ID = %s,
-		    TYPE = IP6_STATIC,
-		    IP6 = %s,
-		    PREFIX_LENGTH = %s,
-		    SIZE = %d ]`
-		return fmt.Sprintf(arstring, arid, arip6, arprefixlength, arsize)
-	}
-	if artype == "IP4_6" {
-		if armac != "" {
-			if argprefix != "" {
-				if arulaprefix != "" {
-					var arstring = `AR = [
-                        AR_ID = %s,
-		                TYPE = IP4_6,
-		                IP = %s,
-		                MAC = %s,
-		                GLOBAL_PREFIX = %s,
-		                ULA_PREFIX = %s,
-		                SIZE = %d ]`
-					return fmt.Sprintf(arstring, arid, arip4, armac, argprefix, arulaprefix, arsize)
-				}
-				var arstring = `AR = [
-                    AR_ID = %s,
-		            TYPE = IP4_6,
-		            IP = %s,
-		            MAC = %s,
-		            GLOBAL_PREFIX = %s,
-		            SIZE = %d ]`
-				return fmt.Sprintf(arstring, arid, arip4, armac, argprefix, arsize)
-			}
-			var arstring = `AR = [
-                AR_ID = %s,
-		        TYPE = IP4_6,
-		        IP = %s,
-		        MAC = %s,
-		        SIZE = %d ]`
-			return fmt.Sprintf(arstring, arid, arip4, armac, arsize)
-		}
-		var arstring = `AR = [
-            AR_ID = %s,
-		    TYPE = IP4_6,
-		    IP = %s,
-		    SIZE = %d ]`
-		return fmt.Sprintf(arstring, arid, arip4, arsize)
-	}
-	if artype == "IP4_6_STATIC" {
-		if armac != "" {
-			var arstring = `AR = [
-                AR_ID = %s,
-		        TYPE = IP4_6_STATIC,
-		        IP = %s,
-		        MAC = %s,
-		        IP6 = %s,
-		        PREFIX_LENGTH = %s,
-		        SIZE = %d ]`
-			return fmt.Sprintf(arstring, arid, arip4, armac, arip6, arprefixlength, arsize)
-		}
-		var arstring = `AR = [
-            AR_ID = %s,
-		    TYPE = IP4_6_STATIC,
-		    IP = %s,
-		    IP6 = %s,
-		    PREFIX_LENGTH = %s,
-		    SIZE = %d ]`
-		return fmt.Sprintf(arstring, arid, arip4, arip6, arprefixlength, arsize)
-	}
-	if artype == "ETHER" {
-		if armac != "" {
-			var arstring = `AR = [
-                AR_ID = %s,
-		        TYPE = ETHER,
-		        MAC = %s,
-		        SIZE = %d ]`
-			return fmt.Sprintf(arstring, arid, armac, arsize)
-		}
-		var arstring = `AR = [
-            AR_ID = %s,
-		    TYPE = ETHER,
-		    SIZE = %d ]`
-		return fmt.Sprintf(arstring, arid, arsize)
-	}
-
-	return ""
+	ar.Add(vnk.ARID, arid)
+	ar.Add(vnk.Size, fmt.Sprint(arsize))
+
+	if armac != "" {
+		ar.Add(vnk.Mac, armac)
+	}
+
+	switch artype {
+	case "IP4":
+		ar.Add(vnk.Type, "IP4")
```

Comment:
all call to `ar.Add(vnk.Type, XXX)` can be done outside the switch with `ar.add-vnk.Type, artype)`